### DISCUSSION TITLE
NATS client-server architecture problem

### DISCUSSION THREAD

**Discussion by michallm** [nats-io/nats-server#2933] [Q&A] [answered]
Hello,

I'm having trouble designing consumers on multiple target devices.

I have one main nats-server (jetstream) that connects multiple devices. One stream is created on the server, let's assume the name ORDERS.

Each device is to receive events from the server. Eg device1 will get orders.created event and device2 too.

Should I create a separate consumer for each topic? For 5 themes, it will be 5 consumers on the server. For two devices, it will be 10 consumers.
![image](https://user-images.githubusercontent.com/64835708/159033564-1c5013ba-ff96-4acd-9417-98903b375da8.png)

Is it possible to subscribe to different topics with the same consumer and process them separately?

**Comment by derekcollison** [comment, marked as answer]
We normally recommend using a single stream and mux all subjects into that and have a separate filtered consumer for each set of messages you want to process independently..

**Reply by michallm** [reply]
if I have two devices, each of them has the OrderCreated consumer, but with different queue groups (dev1, dev2), both will get an event? I'm right?

**Reply by derekcollison** [reply]
Delivery of messages is bound by the consumer state, so if you have two consumers that have same starting stream sequence then those messages will be delivered to each.

**Reply by michallm** [reply]
Ok. Thank you
